src/main/java/com/dre/brewery/BarrelWoodType.java

```diff
@@ -206,4 +256,147 @@ public static BarrelWoodType parse(String string) {
         return fromName(string);
     }
 
+
+    // If another group needs to be added, consider changing getDistance() and stepTowards() with
+    // a breadth-first search algorithm if implementation becomes too complicated.
+    private enum Group {
```

Comment:
This might simplify things, you really only have 2 axises:
- Heat
- Humidity

So why don't you just use those values here, and just measure the distance between two 2D points as a measure of distance between barrel types (can be euklidian distance, manhatan distance, or checkboard distance, your choice).

So for example, let humidity and heat be a positive integer, negative values can just be reserved for the special case, I guess.

Reply:
That's a good idea, latest commit assigns each group a heat (0-3), humidity (0-2), and "oak-ness" (0-1), then uses the manhattan distance (0-5, 6 is not possible). One side effect of needing to always have a wood type assigned to a brew is that moving from `TEMPERATE`, `HOT_DRY`, and `HOT_HUMID` takes two steps instead of one before.